Flaky test ServletTest.testSimpleIdleRead()

**Jetty version(s)**
12+

**Jetty Environment**
ee10

**Description**
See #10945.

The problem is as follows:

The last request does not send the content, so the Servlet is blocked in a read.
The idle timeout fires, and runs the demand callback, which reads the failure, which wakes up the blocked by throwing, which is then caught and handled in `ServletChannel.handleException()`, resulting in a `sendError()` that succeeds.

In the process above, however, we do not notify the underlying `HttpStream` that there was a failure.
So in case of `HttpStreamOverHTTP1`, the underlying `EndPoint` remains fill interested (because the request was waiting for content that was never sent).

When the `sendError()` completes successfully, `HttpStreamOverHTTP1` is succeeded, where it checks if it was fill interested, finding that it was, and aborting with "read pending".

This is still failing on CI.

Most recent on - https://jenkins.webtide.net/blue/organizations/jenkins/jetty.project/detail/PR-14289/1/tests

```
[ERROR] Tests run: 6, Failures: 1, Errors: 0, Skipped: 0, Time elapsed: 6.078 s <<< FAILURE! -- in org.eclipse.jetty.ee10.servlet.ServletTest
[ERROR] org.eclipse.jetty.ee10.servlet.ServletTest.testSimpleIdleRead -- Time elapsed: 1.016 s <<< FAILURE!
java.lang.AssertionError: 

Expected: is <200>
     but: was <500>
	at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:20)
	at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:6)
	at org.eclipse.jetty.ee10.servlet@12.1.6-SNAPSHOT/org.eclipse.jetty.ee10.servlet.ServletTest.testSimpleIdleRead(ServletTest.java:154)
	at java.base/java.lang.reflect.Method.invoke(Method.java:565)
	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:511)
	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.tryRemoveAndExec(ForkJoinPool.java:1490)
	at java.base/java.util.concurrent.ForkJoinPool.helpJoin(ForkJoinPool.java:2248)
	at java.base/java.util.concurrent.ForkJoinTask.awaitDone(ForkJoinTask.java:499)
	at java.base/java.util.concurrent.ForkJoinTask.join(ForkJoinTask.java:666)
	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:511)
	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1450)
	at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:2019)
	at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:187)
```

--- a/jetty-core/jetty-server/src/main/java/org/eclipse/jetty/server/internal/HttpChannelState.java
+++ b/jetty-core/jetty-server/src/main/java/org/eclipse/jetty/server/internal/HttpChannelState.java
@@ -655,6 +655,7 @@ public class HttpChannelState implements HttpChannel, Components
                 listener.onRequestEnd(_request);
 
             // This is THE ONLY PLACE the stream is succeeded or failed.
+            LOG.atDebug().setCause(failure).log("completing the stream of {}", this);
             if (failure == null)
                 stream.succeeded();
             else

--- a/jetty-ee10/jetty-ee10-servlet/src/test/java/org/eclipse/jetty/ee10/servlet/ServletTest.java
+++ b/jetty-ee10/jetty-ee10-servlet/src/test/java/org/eclipse/jetty/ee10/servlet/ServletTest.java
@@ -148,7 +148,7 @@ public class ServletTest
                 Content-Length: 10
                 
                 """;
-            endPoint.addInput(request);
+            endPoint.addInputAndExecute(request);
             endPoint.addInput("1234567890");
             HttpTester.Response response = HttpTester.parseResponse(endPoint.getResponse(false, 5, TimeUnit.SECONDS));
             assertThat(response.getStatus(), is(HttpStatus.OK_200));

